src/box/index.cc

```diff
@@ -476,6 +476,10 @@ int
 iterator_next(struct iterator *it, struct tuple **ret)
 {
 	assert(it->next != NULL);
+	if (!box_check_deadline()) {
+		diag_set(ClientError, ER_SPACE_ITERATION_TIMEOUT);
+		return -1;
+	}
```

Comment:
These functions are supposed to be trivial wrappers around index/iterator methods. The timeout should be checked at the upper level, in `box_select`, etc. Rationale: we might need to iterate over a space without yields internally for some reason. If you put the checks here, it won't be possible.

Reply:
Moved to the upper level